tests/mime.rs

```diff
@@ -0,0 +1,55 @@
+#[macro_use]
+mod utils;
+
+use rand::{rngs::SmallRng, SeedableRng};
+use tempfile::NamedTempFile;
+
+use crate::utils::write_random_content;
+
+#[test]
+/// Makes sure that the files ouch produces are what they claim to be, checking their
+/// types through MIME sniffing.
+fn sanity_check_through_mime() {
+    let temp_dir = tempfile::tempdir().expect("to build a temporary directory");
+    let temp_dir_path = temp_dir.path();
+
+    let test_file = &mut NamedTempFile::new_in(temp_dir_path).expect("to be able to build a temporary file");
+    write_random_content(test_file, &mut SmallRng::from_entropy());
+
+    let formats = [
+        "tar", "zip", "tar.gz", "tgz", "tbz", "tbz2", "txz", "tlz", "tlzma", "tzst", "tar.bz", "tar.bz2", "tar.lzma",
+        "tar.xz", "tar.zst",
+    ];
+
+    let expected_mimes = [
+        "application/x-tar",
+        "application/zip",
+        "application/gzip",
+        "application/gzip",
+        "application/x-bzip2",
+        "application/x-bzip2",
+        "application/x-xz",
+        "application/x-xz",
+        "application/x-xz",
+        "application/zstd",
+        "application/x-bzip2",
+        "application/x-bzip2",
+        "application/x-xz",
+        "application/x-xz",
+        "application/zstd",
+    ];
```

Comment:
I've checked that if some of these breaks, the error will be descriptive enough as to point out exactly which case failed
```
# If we change `application/x-tar` to `application/ERROR`
  left: `"application/x-tar"`,
 right: `"application/ERROR"`', tests/mime.rs:53:9
```

Still, I believe this test could become more descriptive if we used some sort of parametrizing strategy on this function. I know that the [test-case](https://github.com/frondeus/test-case) module can do this, but since we already are using proptest, does prop test introduce some sort of "Fixed parameters list" for this sort of functionality?

```
# How we could do it with test-case:
    #[test_case("tar",  "application/x-tar" )]
    #[test_case("zip", ""application/zip")]
    ...
    #[test_case("tar.zst",  "application/zstd")]
    fn sanity_check_through_mime(format: &str, expected_mime: &str) {
        ...
    }
```

I'll already extend this question to the other mantainers: If proptest can't do it, I'm fine with leaving this test as it is to avoid bringing `test-case` as a dev-dependency, do you agree?

Reply:
I didn't do anything to this test except for removing the no longer needed sleep because I didn't know what to do about it
I think it would be a little confusing to use test_case and proptest at the same time, but we can probably implement a trait to convert `Extension` to mime types and integrate proptest with that

Reply:
Yea, I agree with you that we don't need to do that right now, since it's not the focus of your PR and it was already this way beforehand.